Mathlib/RingTheory/UniqueFactorizationDomain.lean

```diff
@@ -1580,12 +1584,15 @@ theorem prod_le [Nontrivial α] {a b : FactorSet α} : a.prod ≤ b.prod ↔ a 
         prod_mono
 #align associates.prod_le Associates.prod_le
 
+open Classical in
 noncomputable instance : Sup (Associates α) :=
   ⟨fun a b => (a.factors ⊔ b.factors).prod⟩
 
+open Classical in
 noncomputable instance : Inf (Associates α) :=
```

Comment:
When fixing the "unused DecidableEq" warnings I ended up noticing that there were more unnecessary `DecidableEq` assumptions in this file. Now the diff has increased significantly, so you might want to review again